### DISCUSSION TITLE
appsettings.json layering

### DISCUSSION THREAD

**Discussion by steveoh** [dotnet/yarp#725] [Q&A] [answered]
I've noticed that I can split my `ReverseProxy` appsettings between the base and the environments. This is great as it has allowed me to put my routes in the base and the clusters in the environments for deployments. 

I haven't been able to go one level further and split the routes between the appsettings to have them merge back together correctly. Is this possible or am I fouling up the syntax?

**Comment by Tratcher** [comment, marked as answer]
I think the limitation here is that routes are listed in the config as an array and when you nest config files it doesn't merge them quite like you'd expect. See https://github.com/dotnet/extensions/issues/3083.

**Reply by steveoh** [reply]
ok so it's not possible. thanks!
